### DISCUSSION TITLE
HyperHDR 21.0.0.0beta1

### DISCUSSION THREAD

**Discussion by awawa-dev** [awawa-dev/HyperHDR#1011] [Q&A]
I have the pleasure of submitting another version for testing: HyperHDR 21 beta1. I hope you will find the new features & improvements interesting and useful.

 - The architecture has been rebuilt once again, this time focusing on further eliminating heavy dependencies from the past. In addition to lower resource consumption, you will definitely notice how much the installers have been slimmed down (they still contain all the necessary dependent components) and the HyperHDR application itself.

- The DirectX grabber has been expanded with a multi-monitor option. You no longer have to use external applications to capture a second or third monitor. Do you still use them? So please make sure they provide full acceleration support by the graphics card as HyperHDR does including using pixel & vertex shaders to scale the textures before sending them from the graphics card memory to the computer so as not to block communication resources which can cause for example: performance drops in games. It is also the cheapest and most effective alternative to HDMI grabbers and splitters on Windows x64 platforms such as multimedia miniPCs with N100.  
<br/>More about DirectX 11 grabber can be found here: https://github.com/awawa-dev/HyperHDR/wiki/DirectX-11-software-grabber

- Another new feature that I think will be useful for many people is automatic tone mapping. In case you can't control it from outside via the application's API this is the solution for you. Once enabled HyperHDR will analyze the video image and enable it automatically for dark and faded raw HDR signal and disable it when it detects bright SDR image with normal colors.  
<br/> More about automatic tone mapping can be found here: https://github.com/awawa-dev/HyperHDR/wiki/Automatic-tone-mapping
<br/>**If you own Elgato HD60X make sure you read it ⚠️ ⇒** https://github.com/awawa-dev/HyperHDR/discussions/1010

- In the past, the process of calibrating the grabber's LUT to obtain the best possible colors was quite problematic: to do it, it usually required a PC set. HyperHDR v21 introduces a small revolution here because now all you have to do is play the calibration MP4 file in your favorite player and HyperHDR will take care of the rest. Thanks to this, we calibrate not only the grabber and TV, but also your default video player. HyperHDR was also the first to introduce support for the NV12 codec for flatbuffers, which opened completely new perspectives for the calibration of external systems because standard RGB flatbuffer stream was usually already processed at the source.  
<br/>You can find more about the new LUT calibration using MP4 files here: https://github.com/awawa-dev/HyperHDR/wiki/lut-calibration

- I am also pleased to announce for the first time support for the 10-bit P010 codec, which is already appearing in increasingly cheaper grabbers such as the UGreen 25173 (which can only gain in attractiveness after what happened with ELgato HD60X, more: https://github.com/awawa-dev/HyperHDR/discussions/1010). While it does not contribute much to SDR signals, it changes the situation a lot when it comes to HDR by expanding the default narrow bandwidth range above the standard 8-bits, minimizing the quantization effect. Another advantage is that it is usually a raw signal not damaged by the grabber trying to process the HDR signal as SDR. The format is supported by default by Windows, and when it comes to Linux, I created a patch for the UVC module so that it also works on this system.  
<br/>The HyperHDR image already has a patch for P010. Note that it is based on the newer Bookworm so configuring wifi via wpa_supplicant will no longer work: for this you need to use the Raspberry PI imager installer (https://github.com/raspberrypi/bookworm-feedback/issues/72). For Raspberry Pi 3b and Zero 2W users, I would like to remind you about a bug in the Raspberry Pi kernel that can cause incorrect display of colors on Bookworm (https://github.com/awawa-dev/HyperHDR/discussions/848).  
<br/>More about support for the P010 format can be found here: https://github.com/awawa-dev/HyperHDR/wiki/p010-video-format

Full changelog:
- Automatic tone mapping (#998)
- Add Ubuntu 24.10 and Fedora 41 (#1001)
- Add support for "unicam image" RPi HDMI capture card and UYVY format (#889)
- Breaking change: remove LUT from installer and create it dynamically (#994)
- Fix macOS 15 dark theme menu icons (#988)
- Add P010 HDR10 video format support (#968)
- DirectX multi-monitor support (#966)
- macOS 15 sequoia: migrate to ScreenCaptureKit (#984)
- Add Skydimo support (#985)
- New LUT calibration based on mp4 test videos (part I) (#896)
- feat: updated rpi_ws281x submodule, thanks @andreasvikke (#974)
- DX grabber: stick to user specified device selection (#961)
- Fix LED colors adjustments by configuration name (#956)
- Add autoresume feature for the macOS software grabber (#879)
- Add NV12 image format support for flatbuffers (#920)
- Update language file's Thanks @AstaRom (#883, #923, #998)
- Update flatbuffers to v24.3.25 (#875)
- Add cache_cleaner for Github Action (#910)
- New build.sh script for the HyperHDR build process (#904)
- Remove Alsa libs from CMake recipe
- Remove more unnecessary libraries from installers
- Remove libasound from installers
- Fix restoring calibration settings (#874)
- Reorganize HyperHDR libs (#887)
- Fix macOS image alignment (#838)
- Remove hyperhdr-remote app. Reason: #856
- Refactoring of the FlatBuffers client and server (#880)
- Remove mbedtls. Use OpenSSL (#877)
- Removal of QT SQL. Migrate to the SQLite library. (#872)
- New smoothing option for sound effects, fix ArchLinux installer, refactoring of the smoothing module and music effects (#871)
- Removal of QT D-Bus, switch to sdbus-cpp (#864)
- Dependency reductions: removal of heavy QT Gui lib (#861)
- Fix DirectX capturing after new Systray changes
- New independed systray. Dependency reductions thanks to removal of QWidget (#852)
- Fix: lut calibration for grabberless builds (#840)

<hr /><em>This discussion was created from the release <a href='https://github.com/awawa-dev/HyperHDR/releases/tag/v21.0.0.0beta1'>HyperHDR 21.0.0.0beta1</a>.</em>

**Comment by Muyagi** [comment]
Hi,

I bought a Ugreen 25173 and didn’t manage to get the LUT calibration done. 

I use a ezcoo SP12H21 splitter with scale down enabled. 

I tried different resolutions with p010 but the calibration fails. 

It ist always waiting for the box. I let the video run 1 time, or does it have to be in a loop?
After aborting it says that the black is to noisy or the resolution is wrong. I think I made a mistake but don’t know where. 

Do you have a little write up? Or, even better a LUT calibration file for the ugreen? :)

Thanks in advance.

**Comment by awawa-dev** [comment]
Hi

Start the video and then after few secs start the calibration. Just like in the tutorial. The video must be playing before starting the calibration.

> ... or the resolution is wrong.

this is something different and that message doesn't lie. Use p010 1080p25 and have 'Quarter of frame mode' disabled.

**Reply by Muyagi** [reply]
As always. Thanks for the fast response. I’ll try it.

**Reply by Muyagi** [reply]
Still won’t work. I made 2 pictures with the preview on. While it’s waiting, color is ok. When it’s looking for the cards, it has this color distortion. Any advice? :)
![IMG_8389](https://github.com/user-attachments/assets/4e418acb-1a69-4492-ad88-5327dd77c11e)
![IMG_8390](https://github.com/user-attachments/assets/540ae401-312b-4e3f-b66f-6629fe69b80d)

**Reply by awawa-dev** [reply]
The image is mirrored horizontally: the dots in the first and last row should be on the left, not right.

EDIT change of colors is normal during calibration.

**Reply by Muyagi** [reply]
I have no clue how that happend. I just downloaded the the file and played from my mobile to the TV. Now it detects the cards but gives me an error.
I can´t thank you enough for your support.

See attached the logfile.
[Logfile.txt](https://github.com/user-attachments/files/18171609/Logfile.txt)

**Reply by awawa-dev** [reply]
How do you capture video played on your TV using a USB Grabber? No screenshots this time? Because the logs clearly show that the test image is severely distorted and you managed to only capture CRC correctly that I pointed earlier? I have no idea how to help you in this case.

**Reply by Muyagi** [reply]
I use a ezcoo SP12H21 splitter with scale down enabled. The IN comes from the receiver, out 1 to the tv. Out 2 goes in the ugreen, which is connected to the pi 5 via usb 3.0. 
The video is streamed via Apple TV. 

I flipped the video horizontal and then it started the calibration. I hadn’t the preview on. So I didn’t take pictures. 

I will test a few things the next days to see if there’s something interfering. I’ll give you an update then.

**Reply by awawa-dev** [reply]
Simply I just don't understand how you can have an inverted image in live preview when using our test MP4 video files and why do you have to invert it this way or that to correct it? What video player offers such an option and where it is available? Is the image also inverted on TV and looks the same as in the grabber preview?

**Reply by Muyagi** [reply]
I downloaded the file to my iPhone. Saved it to the gallery. In the photos app it is possible to flip the video. 

In the gallery I play the video and watch it via airplay on my Apple TV. 

Here you can see the preview in my download folder. There is it how I downloaded it. 

UPDATE:
I downloaded it again, now its correct. 
I test it again, I saw that my flipped video is upside down. 

![IMG_3189](https://github.com/user-attachments/assets/0f1acd0e-0c8c-4ee1-9825-04e4aa7be428)

**Reply by awawa-dev** [reply]
Just play the original MP4 test video on your Apple TV without using the phone to download it or playing it, probable it flipped the video this time vertically. Maybe it depends on the phone orientation.

**Reply by Muyagi** [reply]
Downloaded it again. Now it works like charm.

**Comment by sjl68** [comment]
Awawa-dev can this beta be used with Libreelec or do we need to wait for a new plugin?

Thanks

**Reply by awawa-dev** [reply]
I will prepare a plugin in next few days

**Reply by awawa-dev** [reply]
Done. New v21 plugin for Libreelec 12 is just about ~7MB compared to the v20:  ~47MB 😃 I hope it still works OK.

**Reply by sjl68** [reply]
Thank you - super quick as always! Installed and all working ok apart from a small bug that I referenced in the discussion below. The suspend function in LE keeps the last grabbed screen active.

https://github.com/awawa-dev/HyperHDR/discussions/740

**Reply by awawa-dev** [reply]
Hmm...but the LEDs are turned off when LE is sleeping? The power management script is still there: https://github.com/awawa-dev/LibreELEC.tv/blob/libreelec-12.0-hyperhdr/packages/addons/service/hyperhdr/source/system.d/service.hyperhdr-power-mng.service

**Reply by sjl68** [reply]
No they are not turned off - when I suspend LE they stay active

![image](https://github.com/user-attachments/assets/f8ce735a-c3ad-4c89-9635-3a8b9ac742c1)

**Reply by awawa-dev** [reply]
So probably we need to test these commands again in LE12: https://github.com/awawa-dev/HyperHDR/discussions/740#discussioncomment-8200233
could you ssh to LE and test these commands and see if HyperHDR reacts OK?

**Reply by sjl68** [reply]
@awawa-dev sorry for the delay, I have tested those commands and they work as they should :)

If I revert to the previous plug in it works as expected and releases hyperhdr.

**Comment by EvoSems** [comment]
Just setup my RPI4 - Lolin - Multi Segment again after doing a Reno. 
I ditched the level shifter this time as I could not for the life of me get the earth situation working flawlessly, seems to work well without. ( I can add it again later I guess if issues arise.) 
I updated to the 21 update and wow! love it! Auto HDR mapping is nearly perfect and very impressive addition, and the LUT calibration is just flawless. well done!

**Reply by awawa-dev** [reply]
If it ain't broke, don't fix it 😉 In some circumstances, with favorable winds after tests, you can do without a level shifter. But if something starts happening, give the Adafruit Scorpio rp2040 a chance: it has built-in level shifter and supports multi-segment in HyperSerialPico as well as in HyperSPI.

**Reply by EvoSems** [reply]
thanks for the rp2040 idea, I think ill try that! 
I Just received the UGREEN 25173 just now. So far with no splitter CEC works as well as HDCP has no issue. well I just watch some shows on Netflix (A.Carbon) no issues. let me know how to properly test this though.

Damn... the negative.... No Dolby Vision

**Reply by awawa-dev** [reply]
Yes I warned about it https://github.com/awawa-dev/HyperHDR/discussions/967#discussion-7403014 If we could change the EDID to support LLDV and flash the firmware back but there is no firmware tools available 😞

**Reply by Muyagi** [reply]
> thanks for the rp2040 idea, I think ill try that! I Just received the UGREEN 25173 just now. So far with no splitter CEC works as well as HDCP has no issue. well I just watch some shows on Netflix (A.Carbon) no issues. let me know how to properly test this though.
> 
> Damn... the negative.... No Dolby Vision

On the Ezcoo splitter, set the EDID Manual to Dolby Vision. This works for me. I can grab DV in correct colors.

**Reply by awawa-dev** [reply]
Yes, Ugreen still requires Ezcoo for LLDV or HDCP.

**Reply by EvoSems** [reply]
yes funnily enough if I use my UGreen splitter before the grabber I can get LLDV lol.

ok. any ideas what could pull the firmware program or app? 
I will have a go best I can before my return window expires to pull the firmware. I'll find my serial clip. 
the hold reset and plug in usb doesn't seem to trigger dfu mode.

I also think the usb power of the grabber pulls more power than the rpi can supply. maybe some kind of injection might be needed.

**Reply by awawa-dev** [reply]
Which RPi and power supply? My UGreen works just fine for many hours without any break.

**Reply by EvoSems** [reply]
I have the firmware dump. now what? can I send to you?

**Reply by awawa-dev** [reply]
How did you manage to do that?!

**Reply by EvoSems** [reply]
well I hope I have something useful. I'm not to sure. I pulled from the chip.
![IMG_4800](https://github.com/user-attachments/assets/03aee259-00ca-473a-acd9-4142d1b6e629)

**Reply by awawa-dev** [reply]
Please send it to my email (in my profile). I will try to locate EDID table. But from my experience with MS2131 changed EDID worked fine until you connect an external HDMI, then EDID was built dynamically which it beyond our reach. Also usually firmware has a checksum, sometimes it is important only for the flasher but sometimes chip may also perform validation.
Did you notice main chipset on the board?

**Reply by EvoSems** [reply]
I have sent. but tbh it looks all gibberish. there is a second chip which I cannot read. (Atmel 24C02N)

edit. managed to read 2nd chip over I2C. its just all FF FF FF

I will have to leave you guys here. I have a lot of home projects I need to get to around the house. 
I have reverted back to the ms2130 for now. it's pretty rock solid. thanks
